lib/i18n/backend/base.rb

```diff
@@ -241,9 +241,9 @@ def load_rb(filename)
         def load_yml(filename)
           begin
             if YAML.respond_to?(:unsafe_load_file) # Psych 4.0 way
-              YAML.unsafe_load_file(filename)
+              YAML.unsafe_load_file(filename, freeze: true, symbolize_names: true)
             else
-              YAML.load_file(filename)
+              YAML.load_file(filename, freeze: true, symbolize_names: true)
```

Comment:
So it fail on older rubies because these options weren't added that long ago:

```
 #<ArgumentError: unknown keywords: freeze, symbolize_names>
```

So there's two ways out of this:

  - You can only do it on `unsafe_load_file`, there's only a couple versions that have `freeze` and `symbolize_names` but not `unsafe_load_file`
  - Or you can feature test each, e.g:
```ruby
YAML_LOAD_OPTIONS = {}
begin
  YAML.load("", symbolize_names: true)
  YAML_LOAD_OPTIONS[:symbolize_names] = true
rescue ArgumentError
end

begin
  YAML.load("", freeze: true)
  YAML_LOAD_OPTIONS[:freeze] = true
rescue ArgumentError
end

def load_yml(filename)
  ...
  YAML.load_file(filename, **YAML_LOAD_OPTIONS)
```

Reply:
Right, I was wondering how this was normally handled! Thanks for sharing, I've _mostly_ adopted your second suggestion 👍